### PR TITLE
Enable delete board

### PR DESCRIPTION
**Pull request by mayalmog**
Popovers for close-board and leave-board created.
Boards can only be closed by their "createdBy" user.
+workspace page re-renders on boards change.

### CODE REVIEW

File: frontend/src/cmps/popovers/popover-close-board.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  const board = useSelector((state) => state.boardReducer.board);
+  const loggedinUser = userService.getLoggedinUser();
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const onCloseBoard = (ev) => {
+    ev.preventDefault();
+    if (loggedinUser.fullname === board.createdBy.fullname) {
+      dispatch(onRemoveBoard(board));
```
Comment (zivbryk) [inline frontend/src/cmps/popovers/popover-close-board.jsx:19]: send boardId instead the whole board object

File: frontend/src/cmps/popovers/popover-create-board.jsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { openPopover } from "../../store/actions/app.actions";
 import { closePopover } from "../../store/actions/app.actions";
 
 export const PopoverCreateBoard = ({ elPos, handleClose }) => {
-  const board = useSelector((state) => state.boardReducer.board);
+  // const board = useSelector((state) => state.boardReducer.board);
```
Comment (zivbryk) [inline frontend/src/cmps/popovers/popover-create-board.jsx:15]: remove comment or add "todo"

File: frontend/src/cmps/popovers/popover-leave-board.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  const onLeaveBoard = (ev) => {
+    ev.preventDefault();
+    if (
+      board.boardMembers.some(
+        (member) => member.fullname === loggedInUser.fullname
+      )
+    ) {
+      let boardToSave = _.cloneDeep(board);
+      const memberIdx = boardToSave.boardMembers.findIndex(
```
Comment (zivbryk) [inline frontend/src/cmps/popovers/popover-leave-board.jsx:26]: check if needed to also remove user from all cards.

File: frontend/src/cmps/popovers/popover-menu.jsx [hunk 4]
```diff
@@ -60,6 +61,29 @@ export const PopoverMenu = ({ elPos, handleClose, board }) => {
             </button>
           </li>
 
+          {board.boardMembers.some(
+            (member) => member.fullname === loggedInUser.fullname
```
Comment (zivbryk) [inline frontend/src/cmps/popovers/popover-menu.jsx:65]: move to separate function

File: frontend/src/pages/board-page.jsx [hunk 5]
```diff
@@ -18,6 +18,8 @@ import { closePopover } from "../store/actions/app.actions";
 
 export const BoardPage = () => {
   const board = useSelector((state) => state.boardReducer.board);
+  // const loggedInUser = useSelector((state) => state.userReducer.loggedInUser);
```
Comment (zivbryk) [inline frontend/src/pages/board-page.jsx:21]: remove comment

File: frontend/src/store/actions/board.actions.js [hunk 6]
```diff
@@ -42,3 +42,16 @@ export function onEditBoard(boardToSave) {
     }
   };
 }
+
+export function onRemoveBoard(board) {
+  return async (dispatch) => {
+    try {
+      await boardService.remove(board._id);
+      dispatch({ type: "REMOVE_BOARD", board });
+      // showSuccessMsg("Board removed succesfully");
```
Comment (zivbryk) [inline frontend/src/store/actions/board.actions.js:51]: remove comment

**Review by zivbryk — CHANGES_REQUESTED** [review]

File: frontend/src/cmps/popovers/popover-close-board.jsx (hunk 1 above)
Comment (mayalmog) [reply frontend/src/cmps/popovers/popover-close-board.jsx]: Done

File: frontend/src/cmps/popovers/popover-leave-board.jsx (hunk 3 above)
Comment (mayalmog) [reply frontend/src/cmps/popovers/popover-leave-board.jsx:34]: done

File: frontend/src/cmps/popovers/popover-close-board.jsx (hunk 1 above)
Comment (mayalmog) [reply frontend/src/cmps/popovers/popover-close-board.jsx]: Current code: 

export function loadBoard(boardId) {
  return async (dispatch) => {
    try {
      const board = await boardService.getById(boardId);
      dispatch({ type: "SET_BOARD", board });
    } catch (err) {
      console.log("board.actions: err @ loadBoard", err);
    }
  };
}

**Review by zivbryk — APPROVED** [review]

**Pull request merged**